### DISCUSSION TITLE
Improve mesh quality

### DISCUSSION THREAD

**Discussion by JadeUoM** [neperfepx/neper#777] [Q&A] [answered]
Hi Quey,

I hope this message finds you well. 

I am currently working with Neper to generate a mesh for a cylinder using a custom seed file and orientation file. However, I've noticed some issues with the mesh quality, particularly in areas where several grains meet. Additionally, there are instances of tiny grains being surrounded by larger grains, as highlight in the figure.
![Cylinder](https://github.com/neperfepx/neper/assets/102606450/fe5a46c9-f7f4-408c-97f0-83d32d82708a)

The seed and orientation files used can be found in:
https://github.com/JadeUoM/Neper.git 

The command to generate the cylinder:
neper -T -domain "cylinder(15, 25, 36.0)" -n "119" -morphooptiini "coo:file(Seed),weight:file(Weight)" -ori "file(OriEuler,des=euler-bunge)" -oridescriptor euler-bunge -oriformat plain -regularization "1" -statcell "id,vol,diameq" -format "tess,vtk,geo,ori,stl" -o Cylinder

neper -M Cylinder.tess -elttype "tet" -rcl "1" -order "1" -meshqualmin "0.9" -format "msh,vtk,inp,med" -faset all -nset "z0,z1" -mesh3dclreps 0.1

I was wondering if you could provide guidance on how to improve the mesh quality in these problematic areas. Are there specific parameters or adjustments I should consider in the Neper configuration to achieve a more uniform mesh, especially around grain boundaries?

Thank you for your time and assistance.

Kind regards,
Jade

**Comment by rquey** [comment, marked as answer]
You are using regularization (`-reg 1`), which already improves mesh quality significantly.

What you typically need here is to "relax" 1D meshing by increasing the `-pl` value (to e.g. 10).  This will generate quite flat elements, but it will take care of the mesh over-refinements that you surrounded.

**Reply by JadeUoM** [reply]
Many thanks, Quey. I'll try with your suggestion and see how it goes.
